lib/oli_web/controllers/cognito_controller.ex

```diff
@@ -0,0 +1,151 @@
+defmodule OliWeb.CognitoController do
+  use OliWeb, :controller
+  import Oli.HTTP
+  import Oli.Utils
+  import Ecto
+  import Ecto.Changeset
+  require Logger
+
+  alias Oli.Repo
+  alias Oli.Accounts
+  alias Oli.Groups.CommunityAccount
+  alias Oli.Institutions.SsoJwks
+
+  def launch(conn, params) do
+    try do
+      with {:ok, valid_params} <- validate_query_params(params) do
```

Comment:
You could pattern match all the fields used below:
```elixir
{:ok, %{id_token: id_token, community_id: community_id, course_section_id: course_section_id} = valid_params} <- validate_query_params(params)
```

and then replace the places where they are used accordingly.

Also, I think a `case` makes more sense than a `with` clause. Wdyt?